grails-app/controllers/org/pih/warehouse/inventory/InventoryItemController.groovy

```diff
@@ -32,7 +31,6 @@ import grails.core.GrailsApplication
 import java.text.DateFormat
 import java.text.SimpleDateFormat
 
-@Transactional
```

Comment:
On one hand there is probably no other way to make this work, but on the other hand I'm afraid of that this change considering the other methods, especially the saving ones, that do not use the flush, won't work.

With the setting `allow_update_outside_transaction` we will only auto-commit the change (which is also not good) if we use the flush (this is probably why you've added `flush: true` to the create method).

I would also be aware of the `InventoryItem` `publishPersistenceEvent` - without transaction and with auto-commit (`flush: true` on update) we might face some weird exceptions when the `afterUpdate` would be executed. Have you made sure those work? 

I think the best idea (and the "cleanest" one) would be just to live with the exceptions that might be thrown, create tickets and move the other methods to the service layer.